qiskit/ignis/calibrations/schedules.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Pulse Schedule Generation for calibration experiments
+"""
+
+import copy
+import numpy as np
+import qiskit
+import qiskit.pulse as pulse
+import qiskit.pulse.pulse_lib as pulse_lib
+from qiskit.circuit import Gate
+from qiskit.scheduler import schedule_circuit, ScheduleConfig
+
+
+def rabi_schedules(amp_list, qubits, pulse_width, pulse_sigma=None,
+                   width_sigma_ratio=4, system=None, cmd_def=None,
+                   meas_map=None):
+    """
+    Generates schedules for a rabi experiment using a Gaussian pulse
+
+    Args:
+        amp_list (list of floats): List of amplitudes for the Gaussian
+        pulse [-1,1]
+        qubits (list of integers): indices of the qubits to perform a rabi
+        pulse_width: width of gaussian (in dt units)
+        pulse_sigma: sigma of gaussian
+        width_sigma_ratio: set sigma to a certain ratio of the width (use if
+        pulse_sigma is None)
+        system: PulseChannelSpec object
```

Comment:
Could we do away with the need for system if we just used `DriveChannel(qubit)` since `system` is only used in a single place. @lcapelluto  has plans to deprecate `PulseChannelSpec`.

Reply:
Fixed